### PR TITLE
Signin page return

### PR DESCRIPTION
**Pull request by radiansz**

### CODE REVIEW

File: client/routes/signin/components/SigninContent.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+const classes = bem('signin-content');
+
+const SigninContent = () => (
+  <div {...classes(':root')}>
+    <div {...classes('tabs-panel')}>
+      <Tabs>
+        <TabList>
+          <Tab>
+            <div>
```
Comment (zouxuoz) [inline client/routes/signin/components/SigninContent.js]: unnecessary wrapper?

File: client/routes/signin/components/SigninContent.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        <TabPanel>
+          <h2>Tab 3</h2>
+        </TabPanel>
+      </Tabs>
+    </div>
+    <div {...classes('language-panel')}>
+      <LanguageSelect />
+    </div>
+    <div>
+      Нет аккаунта? <a href="https://sendsay.ru/signup.html">Зарегистрируйтесь</a>
```
Comment (zouxuoz) [inline client/routes/signin/components/SigninContent.js]: i18n?

File: client/routes/signin/components/SigninContent.css [hunk 3]
```diff
@@ -0,0 +1,36 @@
+.signin-content {
+  align-items: center;
+  display: flex;
+  flex: 1;
+  flex-direction: column;
+  justify-content: flex-start;
+  padding-top: 50px;
+
+  & * {
```
Comment (zouxuoz) [inline client/routes/signin/components/SigninContent.css]: bad practice

File: client/routes/signin/components/SigninContent.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+  },
+
+  corporate: {
+    id: 'signin-content__corporate',
+    defaultMessage: 'Корпоративный вход',
+  },
+
+  agses: {
+    id: 'signin-content__agses',
+    defaultMessage: 'Agses-карта',
```
Comment (zouxuoz) [inline client/routes/signin/components/SigninContent.js]: AGSES

File: client/components/ui/Button/Button.css [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+.button_color_blue {
+  background-color: var(--dirtblueColor);
+  color: #fff;
+
+  &:hover {
+    background-color: var(--sickblueColor);
+  }
+}
+
+@keyframes progress-bar-stripes {
```
Comment (rosendi) [inline client/components/ui/Button/Button.css]: BEM

File: client/routes/signin/components/SigninContent.js [hunk 6]
```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import { Tab, Tabs, TabList, TabPanel } from 'react-tabs';
```
Comment (zouxuoz) [inline client/routes/signin/components/SigninContent.js]: may be create wrappers for all used react-tabs components in components/ui, style and use it?

File: client/components/ui/Text/Text.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+const classes = bem('input-text');
+
+const Text = ({ meta: { touched, error }, name, input, label, placeholder }) => {
+  const invalid = touched && !!error;
+
+  return (
+    <div {...classes(':root', { invalid })}>
+      {label && <label {...classes('label')} htmlFor={name}>{label}</label>}
+      <input {...classes('body')} {...input} name={name} placeholder={placeholder} />
+      {invalid && <span {...classes('error')}>{error}</span>}
```
Comment (rosendi) [inline client/components/ui/Text/Text.js]: touched && error

File: client/components/ui/Text/Text.js [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+    </div>
+  );
+};
+
+Text.propTypes = {
+  label: PropTypes.string,
+  type: PropTypes.string,
+  placeholder: PropTypes.string,
+  classModifiers: PropTypes.shape({
```
Comment (rosendi) [inline client/components/ui/Text/Text.js]: Remove

File: client/routes/signin/components/SigninContent.css [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+    outline: none;
```
Comment (rosendi) [inline client/routes/signin/components/SigninContent.css]: Move to the global css.

File: client/routes/signin/containers/LanguageSelectContainer.js [hunk 10]
```diff
@@ -0,0 +1,30 @@
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { setLocale } from 'logic/modules/locale';
+import LanguageSelect from '../components/LanguageSelect';
+
+const mapStateToProps = state => ({
+  lang: state.locale,
```
Comment (rosendi) [inline client/routes/signin/containers/LanguageSelectContainer.js]: `locale: state.locale`

File: client/routes/signin/containers/LanguageSelectContainer.js [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+  setLocale,
+};
+
+class LanguageContainer extends Component {
+  static propTypes = {
+    lang: PropTypes.string.isRequired,
+    setLocale: PropTypes.func.isRequired,
+  };
+
+  _onLanguageChange = ({ target: { value } }) => {
```
Comment (rosendi) [inline client/routes/signin/containers/LanguageSelectContainer.js]: `onSetLocale`

File: client/routes/signin/containers/SigninFormContainer.js [hunk 12]
```diff
@@ -0,0 +1,67 @@
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { injectIntl, defineMessages } from 'react-intl';
+import SigninForm from '../components/SigninForm';
+import { signin } from '../modules/signin';
+
```
Comment (rosendi) [inline client/routes/signin/containers/SigninFormContainer.js]: Remove new line

File: client/routes/signin/containers/SigninFormContainer.js [hunk 13]
```diff
@@ -0,0 +1,67 @@
...
+const messages = defineMessages({
+  wrongCredentials: {
+    id: 'signin-form__wrong-credentials',
+    defaultMessage: 'Неправильный логин или пароль',
+  },
+});
+
+class SigninContainer extends Component {
+  static propTypes = {
+    signin: PropTypes.func.isRequired,
```
Comment (rosendi) [inline client/routes/signin/containers/SigninFormContainer.js]: add a new line

File: client/components/AppHeader/AppHeader.js [hunk 14]
```diff
@@ -31,34 +31,46 @@ const messages = defineMessages({
   },
 });
 
-const AppHeader = () => (
-  <header className="app-header">
-    <div className="app-header__container">
-      <AppHeaderLogo />
+const renderNav = (authToken) => {
```
Comment (rosendi) [inline client/components/AppHeader/AppHeader.js]: Rename it as a stateless component `AppNavigation`

File: client/components/AppHeader/AppHeader.js [hunk 15]
```diff
@@ -31,34 +31,46 @@ const messages = defineMessages({
...
+  if (!authToken) {
```
Comment (rosendi) [inline client/components/AppHeader/AppHeader.js]: Don't check it here, check it in `AppHeader`

File: client/components/AppHeader/AppHeader.js [hunk 16]
```diff
@@ -31,34 +31,46 @@ const messages = defineMessages({
...
+    </AppHeaderItem>
+  </nav>);
+};
+
+const AppHeader = ({ authToken }) => (
+  <header className="app-header">
+    <div className="app-header__container">
+      <AppHeaderLogo />
+
+      {renderNav(authToken)}
```
Comment (rosendi) [inline client/components/AppHeader/AppHeader.js]: `authenticated && <AppNavigation`

File: client/components/AppHeader/AppHeader.js [hunk 17]
```diff
@@ -31,34 +31,46 @@ const messages = defineMessages({
...
     </div>
   </header>
 );
 
+AppHeader.propTypes = {
+  authToken: PropTypes.string,
```
Comment (rosendi) [inline client/components/AppHeader/AppHeader.js]: authentication: PropTypes.boolean.isRequired

File: client/components/ui/Forms/Button/Button.css [hunk 18]
```diff
@@ -0,0 +1,44 @@
...
+  white-space: nowrap;
+}
+
+.button_loading {
+  animation: progress-bar-stripes 2s linear infinite;
+  background-image: linear-gradient(45deg, rgba(255, 255, 255, 0.5) 25%, transparent 25%, transparent 50%, rgba(255, 255, 255, 0.5) 50%, rgba(255, 255, 255, 0.5) 75%, transparent 75%, transparent);
+  background-size: 40px 40px;
+}
+
+.button_color_blue {
```
Comment (rosendi) [inline client/components/ui/Forms/Button/Button.css]: Is it a default color for button?

File: client/components/ui/Forms/InputText/InputText.css [hunk 19]
```diff
@@ -0,0 +1,41 @@
+.input-text__body {
+  border: 1px solid color(black a(11%));
+  border-radius: 2px;
+  display: block;
+  font-family: "Open Sans", "Arial", sans-serif;
+  font-size: 13px;
+  height: 30px;
+  line-height: 30px;
```
Comment (rosendi) [inline client/components/ui/Forms/InputText/InputText.css]: Don't use line-height here. It's better to use padding/height

File: client/components/ui/Forms/InputText/InputText.css [hunk 20]
```diff
@@ -0,0 +1,41 @@
...
+
+  &:focus {
+    border-color: #007c99;
+  }
+}
+
+.input-text__label {
+  display: block;
+  font-size: 12px;
+  padding-bottom: 4px;
```
Comment (rosendi) [inline client/components/ui/Forms/InputText/InputText.css]: don't use padding between element use margin instead

File: client/components/ui/Forms/InputText/InputText.js [hunk 21]
```diff
@@ -0,0 +1,30 @@
+import React, { PropTypes } from 'react';
+import { propTypes } from 'containers/ui/Form/Field';
+import bem from 'lib/bem';
+import './InputText.css';
+
+const classes = bem('input-text');
+
+const Text = ({ meta: { touched, error }, name, input, label, placeholder }) => {
+  const invalid = touched && !!error;
```
Comment (rosendi) [inline client/components/ui/Forms/InputText/InputText.js]: Move it to 15 line

File: client/components/ui/Forms/InputText/InputText.js [hunk 22]
```diff
@@ -0,0 +1,30 @@
...
+      {invalid && <span {...classes('error')}>{error}</span>}
+    </div>
+  );
+};
+
+Text.propTypes = {
+  label: PropTypes.string,
+  type: PropTypes.string,
+  placeholder: PropTypes.string,
+  classModifiers: PropTypes.shape({
```
Comment (rosendi) [inline client/components/ui/Forms/InputText/InputText.js]: Remove

File: client/components/ui/Text/Text.css [hunk 23]
```diff
@@ -0,0 +1,41 @@
+.input-text__body {
```
Comment (rosendi) [inline client/components/ui/Text/Text.css]: Rename component to TextInput

File: client/containers/AppHeaderContainer.js [hunk 24]
```diff
@@ -0,0 +1,8 @@
+import { connect } from 'react-redux';
+import AppHeader from 'components/AppHeader';
+
+const mapStateToProps = state => ({
+  authToken: state.auth.token,
```
Comment (rosendi) [inline client/containers/AppHeaderContainer.js]: `authenticated: !!state.auth.token`

File: client/routes/signin/components/SigninContent.css (hunk 3 above)
Comment (rosendi) [inline client/routes/signin/components/SigninContent.css]: very bad

File: client/components/AppHeader/AppHeader.js (hunk 17 above)
Comment (radiansz) [inline client/components/AppHeader/AppHeader.js]: It's not required cause when there is no token there is no nav

File: client/routes/signin/components/SigninContent.js [hunk 25]
```diff
@@ -0,0 +1,69 @@
...
+            <FormattedMessage {...messages.agses} />
+          </div>
+        </div>
+
+        <div>
+          <SigninForm />
+        </div>
+      </div>
+    </div>
+    <div {...classes('language-panel')}>
```
Comment (rosendi) [inline client/routes/signin/components/SigninContent.js]: new line

File: client/routes/signin/components/SigninContent.js [hunk 26]
```diff
@@ -0,0 +1,69 @@
...
+      <LanguageSelect />
+    </div>
+    <div>
```
Comment (rosendi) [inline client/routes/signin/components/SigninContent.js]: new line

File: client/routes/signin/components/SigninForm.js [hunk 27]
```diff
@@ -1,49 +1,70 @@
...
-        <label htmlFor="password">Password</label>
-        <Field name="password" type="password" component={Text} />
-        <div>error</div>
+
+      <div {...classes('field')}>
+        <Field name="password" component={InputText} type="password" placeholder={intl.formatMessage(messages.passwordPlaceholder)} />
+      </div>
+
+      <div {...classes('footer')}>
+        <div {...classes('error')} />
```
Comment (rosendi) [inline client/routes/signin/components/SigninForm.js]: new line

File: client/routes/signin/components/SigninForm.js [hunk 28]
```diff
@@ -1,49 +1,70 @@
...
 );
 
-Button.propTypes = {
-  submitting: PropTypes.bool,
-};
-
 SigninForm.propTypes = {
-  signin: PropTypes.func,
+  submit: PropTypes.func.isRequired,
+  onError: PropTypes.func.isRequired,
```
Comment (rosendi) [inline client/routes/signin/components/SigninForm.js]: new line

File: client/routes/signin/containers/LanguageSelectContainer.js [hunk 29]
```diff
@@ -0,0 +1,30 @@
...
+  setLocale,
+};
+
+class LanguageContainer extends Component {
+  static propTypes = {
+    locale: PropTypes.string.isRequired,
+    setLocale: PropTypes.func.isRequired,
+  };
+
+  onLocaleChange = ({ target: { value } }) => {
```
Comment (rosendi) [inline client/routes/signin/containers/LanguageSelectContainer.js]: `onChangeLocale`

File: client/routes/signin/containers/SigninFormContainer.js [hunk 30]
```diff
@@ -0,0 +1,67 @@
...
+  static contextTypes = {
+    router: PropTypes.object.isRequired,
+    location: PropTypes.object.isRequired,
+  };
+
+  onError() {
+    const { intl: formatMessage } = this.props;
+
+    return {
+      _general: formatMessage(messages.wrongCredentials),
```
Comment (rosendi) [inline client/routes/signin/containers/SigninFormContainer.js]: `_error`

File: client/routes/signin/containers/SigninFormContainer.js [hunk 31]
```diff
@@ -0,0 +1,67 @@
+import React, { Component, PropTypes } from 'react';
```
Comment (rosendi) [inline client/routes/signin/containers/SigninFormContainer.js]: Merge that container into SigninContainer

File: client/routes/signin/components/SigninContent.js [hunk 32]
```diff
@@ -0,0 +1,69 @@
+import React from 'react';
```
Comment (rosendi) [inline client/routes/signin/components/SigninContent.js]: Raname it to `Signin.js`

**Pull request merged**